map.go

```diff
@@ -191,6 +191,26 @@ func (m Map) Reduce(f func(acc, key, val interface{}) interface{}, acc interface
 	return acc
 }
 
+// Merge returns a map from the merging between two maps, should there be a key overlap,
+// the value that corresponds to key will be replaced by the value resulted from the
+// provided resolve function.
+func (m Map) Merge(n Map, resolve func(key, a, b interface{}) interface{}) Map {
+	if m.IsEmpty() || m.Count() == 0 {
```

Comment:
Those two tests mean the same thing. Use `IsEmpty()` since it's more specific and potentially more efficient.

Reply:
changed added